Feature/landing v2 2d

luxumbra opened the pull request:
## Overview

**What features/fixes does this PR include?**
This contains my first pass at the 2d version of [the new design](https://www.figma.com/file/n36ggiRBQEGzRUbGxWfkMX/Meta-Game-(Copy)?node-id=5238%3A11728). 
- ~~No responsive has been done, *this is done on desktop only so far*~~
- There was no mobile designs to work from so improvised once main desktop styles were implemented. All devices should look pretty good now but there are areas that probs need revisiting & tidying. 
- Section 5 is missing (needs to be thought about and built)
- Intro section mostly done. Needs feedback on the quote that I added as an 'easter egg' - good solution or UX boo-boo?
- Menu needs styling as per the design.

**Please provide the GitHub issue number**
#938 

## Follow up Improvement Ideas
- [x] Optimise images. 
- [ ] Refactor components, re-use where I can & clean up code.
- [ ] Finish responsive bits
- [ ] Add in some subtle webgl w/threejs and have some subtle mouse/scroll controlled particles floating around in the background. Need transparent background on images for that though. Had thought about applying the images to a plane geometry in threejs, then it makes it easier to apply some parallax and depth related effects but is definitely a *nice-to-have*
- [ ] The above could be a solution to the horizontal scroll functionality of section 5.


**Side effects**
Should only affect the landing page.

vidvidvid commented:
omg this looks so nice 🤤

tommymaher15 commented:
This is incredible!

vidvidvid commented on packages/web/components/Container.tsx:
```diff
@@ -15,6 +15,52 @@ export const PageContainer: React.FC<FlexProps> = ({ children, ...props }) => (
   </Flex>
 );
 
+// export const ScrollContainer: React.FC<FlexProps> = ({ children, ...props }) => {
```
If this is not used anywhere anymore - we could just delete it.

luxumbra replied:
Will remove.

vidvidvid commented on packages/web/components/Landing/Intro.tsx:
```diff
@@ -1,90 +1,353 @@
         </Stack>
+        <ScrollLink showQuote={showQuote} toggleQuote={toggleQuote} />
+        <QuoteLayer
+          quote="You never change things by fighting the existing reality. To change something, build a new model that  makes the existing model obsolete."
```
'that  makes' has an extra space

luxumbra replied:
eagle-eye 🦅

vidvidvid replied:
🦅 :D

vidvidvid commented on packages/web/components/Landing/Intro.tsx:
```diff
@@ -1,90 +1,353 @@
-    },
-    [router],
-  );
+  // const slideStyles = useSpring({
+  //   config: config.stiff,
+  //   from: { transform: `translate3d(0, ${300}px, 0)` },
+  //   to: {
+  //     transform: `translate3d(0, ${showQuote ? -50 : 300}px, 0)`
+  //   }
+  // })
+
+  // const handleSectionNav = useCallback(
+  //   (sectionId: string) => {
+  //     router.push(`#${sectionId}`);
+  //   },
+  //   [router],
+  // );
```
if these values are not needed anymore, why not delete them? could they become useful anytime in the future?

luxumbra replied:
That is the reason I leave things commented out as the commented code doesn't add to the build size but it *does* make the component messy and harder to read. Will remove. 🙏🏻

vidvidvid commented on packages/web/components/Landing/LandingHeader.tsx:
```diff
@@ -90,51 +98,48 @@ export const LandingHeader: React.FC = () => {
-            <MenuIconSVG />
-            <MenuIconSVG />
-            <MenuIconSVG />
+            <MenuIcon2SVG toggle={toggle} />
```
nice :D it looked really funny with the 3 svgs hehe

vidvidvid commented on packages/web/components/Landing/WhatDo.tsx:
```diff
@@ -0,0 +1,234 @@
+  const onScreen = useOnScreen(ref);
+  const section = 'what-do';
+
+  // const createSlidesRefs = (slide: HTMLDivElement, index: number) => {
+  //   slides.current[index] = slide
+  //   console.log('slides', slides);
+
+  // }
+
+  // useEffect(() => {
+  //   const totalSlides = slides.current.length
+
+  //   gsap.to(slides.current, {
+  //     xPercent: 0 * (totalSlides - 1),
+  //     ease: 'none',
+  //     scrollTrigger: {
+  //       trigger: slidesContainer.current,
+  //       pin: true,
+  //       scrub: 1,
+  //       snap: 1 / (totalSlides - 1),
+  //       end: () => `+=${slidesContainer.current?.offsetWidth}`
+  //     }
+  //   })
+  // }, []);
```
more commented out code. why not just delete it?

luxumbra replied:
lol kept for possible future use but kk I'll delete. 😄

vidvidvid replied:
well, stuff like that might be useful. if you decide to leave it in, it would imo be nice to write a short explanation of why it's still there and what was/could be the purpose of it.

luxumbra replied:
For sure. Will do in future but I've now removed it. :)

vidvidvid reviewed (commented):
the landing page keeps on getting more and more epic with every iteration! apart from some minor possible improvements, i have no real suggestions code - wise. still need to click around and check if everything works

luxumbra commented:
@vidvidvid I addressed your comments in my latest update, plus a few others based on feedback from @peth-yursick & @betiiiiiiiiiiiii in Discord.  Thanks for the prompt review ser. 🙏🏻

I changed the `scroll-snap-type` to `y proximity` on mobile to make the scroll snaping less harsh on mobile devices.
Also added a couple of new breakpoints to enable more flexibility above 1536px  (peth is on a 2k display and things needed tweaking beyond 1536 & 1920)

vidvidvid reviewed (approved):
Looks good! Thank you for the prompt changes as well, ser! :D
